console/v1/types_console_plugin.go

```diff
@@ -0,0 +1,197 @@
+	// in case the proxied Service is using custom service CA.
+	// By default, the service CA bundle provided by the service-ca operator is used.
+	// +kubebuilder:validation:Pattern=`^-----BEGIN CERTIFICATE-----([\s\S]*)-----END CERTIFICATE-----\s?$`
+	// +kubebuilder:validation:Optional
+	CACertificate string `json:"caCertificate,omitempty"`
+	// authorization provides information about authorization type,
+	// which the proxied request should contain
+	// +kubebuilder:validation:Optional
+	Authorization AuthorizationType `json:"authorization,omitempty"`
+}
+
+// ConsolePluginProxyEndpoint holds information about the endpoint to which
+// request will be proxied to.
+type ConsolePluginProxyEndpoint struct {
+	// type is the type of the console plugin's proxy. Currently only "Service" is supported.
+	// +kubebuilder:validation:Required
+	Type ConsolePluginProxyType `json:"type"`
+	// service is an in-cluster Service that the plugin will connect to.
+	// The Service must use HTTPS. The console backend exposes an endpoint
+	// in order to proxy communication between the plugin and the Service.
+	// Note: service field is required for now, since currently only "Service"
+	// type is supported.
+	// +kubebuilder:validation:Required
+	Service ConsolePluginProxyServiceConfig `json:"service"`
+}
+
+// ProxyType is an enumeration of available proxy types
+// +kubebuilder:validation:Pattern=`^(Service)$`
+type ConsolePluginProxyType string
+
+const (
+	// ProxyTypeService is used when proxying communication to a Service
+	ProxyTypeService ConsolePluginProxyType = "Service"
+)
+
+// AuthorizationType is an enumerate of available authorization types
+// +kubebuilder:validation:Pattern=`^(UserToken|None)$`
+// +kubebuilder:default:="None"
+type AuthorizationType string
+
+const (
+	// UserToken indicates that the proxied request should contain the logged-in user's
+	// OpenShift access token in the "Authorization" request header. For example:
+	//
+	// Authorization: Bearer sha256~kV46hPnEYhCWFnB85r5NrprAxggzgb6GOeLbgcKNsH0
+	//
+	UserToken AuthorizationType = "UserToken"
+	// None indicates that proxied request wont contain authorization of any type.
+	None AuthorizationType = "None"
+)
+
+// ProxyTypeServiceConfig holds information on Service to which
+// console's backend will proxy the plugin's requests.
+type ConsolePluginProxyServiceConfig struct {
+	// name of Service that the plugin needs to connect to.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	Name string `json:"name"`
+	// namespace of Service that the plugin needs to connect to
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	Namespace string `json:"namespace"`
+	// port on which the Service that the plugin needs to connect to
+	// is listening on.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Maximum:=65535
```

Comment:
Can services expose ports that high?

Reply:
Havent found any restriction in the port's range. Also other port fields are using this rang
<img width="570" alt="Screenshot 2022-08-01 at 16 13 34" src="https://user-images.githubusercontent.com/1668218/182167453-315cf77b-807b-439e-8cb3-f0a1417dad0d.png">